### PR TITLE
Generic entities reducer

### LINKED ISSUE #15: Add actions and reducer to load and set tags, clients, projects and tasks
For each entity we will need the following actions:
- loadWhatever
- setWhatever (with an array of that entity as payload)

And a single reducer for each of those to handle those actions. Also handle errors and add tests.

### PR DESCRIPTION
**Pull request by Odrakir**
Closes #15 

This PR is a bit more complex, because I did some refactoring to create a generic entity reducer, so we don't have to repeat all the operations (set, insert, delete...)

Commit history is clean, so this can be reviewed one commit at a time.

As this PR is long, I'm creating other issues for error handling and tests.

### CODE REVIEW

File: TogglWatch.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,6 +7,8 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		8C00AF6B238412A6001FB22F /* AppReducer.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8C00AF692384120C001FB22F /* AppReducer.swift */; };
+		8C00AF6C238412A7001FB22F /* AppReducer.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8C00AF692384120C001FB22F /* AppReducer.swift */; };
```
Comment (sofokli1) [inline TogglWatch.xcodeproj/project.pbxproj]: this looks like an other artefact

File: TogglWatch WatchKit Extension/ContentView.swift [hunk 2]
```diff
@@ -51,9 +56,9 @@ struct ContentView: View
                         action: { .timelineEntries($0)}
                     )
                 )
-                .onAppear {
-                    self.store.send(.user(.loadAPITokenAndUser))
-                    self.store.send(.loadAll)
+                    .onAppear {
```
Comment (sofokli1) [inline TogglWatch WatchKit Extension/ContentView.swift]: why the indentation?

File: TogglWatch.xcodeproj/project.pbxproj (hunk 1 above)
Comment (Odrakir) [reply TogglWatch.xcodeproj/project.pbxproj]: I think this is due to those files being in 2 targets. It happens with all the TogglTrack files.

File: TogglWatch WatchKit Extension/ContentView.swift (hunk 2 above)
Comment (Odrakir) [reply TogglWatch WatchKit Extension/ContentView.swift]: For whatever reason, this is how Xcode thinks it's supposed to look. I'm fixing it.

**Pull request merged**